nixos/modules/programs/sway-beta.nix

```diff
@@ -0,0 +1,33 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.sway-beta;
+  swayPackage = cfg.package;
+in {
+  options.programs.sway-beta = {
+    enable = mkEnableOption ''
+      The i3-compatible Wayland compositor. (temporary beta package)
```

Comment:
The description isn't that relevant, but if possible you should start with a lowercase character (the result will be "Whether to enable <TEXT>.") and package might be a bit misleading here.

Maybe something like: "Sway, the i3-compatible tiling Wayland compositor. This module will be removed after the final release of Sway 1.0"?

Other than that it looks good to me :)